src/drive/actions/async.js

```diff
@@ -71,64 +75,81 @@ class Stack {
     sortAttribute,
     sortOrder = 'asc',
     skip = 0,
-    limit = FILES_FETCH_LIMIT,
-    loadedFoldersCount = 0,
-    loadedFilesCount = 0
+    limit = FILES_FETCH_LIMIT
   ) => {
-    const shouldSeparateFilesAndFolders = sortAttribute === 'name'
-    const index = await cozy.client.data.defineIndex('io.cozy.files', [
+    const isFirstLoad = skip === 0
+    let folders = []
+
+    if (isFirstLoad) {
+      const folderSortingOrder = sortAttribute === 'name' ? sortOrder : 'asc'
+      const allFolders = await this.query({
+        folderId,
+        type: 'directory',
+        sortAttribute,
+        sortOrder: folderSortingOrder,
+        skip: 0,
+        limit: null
+      })
+      folders = allFolders.filter(folder => folder._id !== TRASH_DIR_ID)
```

Comment:
The first time the view is loaded, we load *all* the folders — then when the user scrolls and we fetch more data, we can focus on the files. If there are tons of folders this might end up being slower, but that should be rare.

We also filter out the trash folder with javascript as it is faster than doing it through Pouch.

Reply:
I think these 2 informations deserve a little comment in the code, don't they?